### DISCUSSION TITLE
Jess Martin - DXOS

### DISCUSSION THREAD

**Discussion by just-be-dev** [devtools-fm/devtools.fm#223] [Announcements]
## Background

- https://twitter.com/jessmartin
- https://jessmart.in/
- https://twitter.com/dxos_org
- https://dxos.org/

## Intro

**Andrew** 

Hello and welcome to the devtools.fm podcast. This is a podcast about developer tools and the people who make them. I'm Andrew and this is my co-host Justin.

**Justin** 

Hey everyone! Our guest today is Jess Martin who I've had the absolute pleasure of getting to know through various meetups and conferences we've run into each other at. Jess, you're currently working with DXOS which we're excited to talk more about, but you've also done a _lot_ of other things. Before we dive in, would you like to tell our listeners a little more about yourself?

## Topics

- background
- dxos
- local-first apps
- composer (https://www.youtube.com/watch?v=xN0ZJHN_k2c)

### Background

- [x] You've been a successful founder and now spend time consulting with various companies working on topics you're interested in. How has that transition been and why do you choose to work in the way you do now?
- [x] You're very active in the local-first, future of computing, and end-user programming spaces. What draws you to these areas?

### DXOS

- [x] What is DXOS? 
- [x] When would you choose to use it and when would you choose not to use it?
- [x] When should someone choose a peer-to-peer framework like DXOS over a more centralized service?
- [x] DXOS uses ECHO database, HALO Identity, MESH Networking, and KUBE Network. What are these components and how do they work together?
- [x] How does ECHO differ from traditional databases? 
- [x] Are there types of data that ECHO wouldn't be good at storing?
- [x] Why was a custom identity system like HALO required for DXOS?
- [ ] How does KUBE Networking fit into the overall DXOS scheme?
- [ ] What functionality does the developer tools provide?

### Composer

- [x] You've recently been working on a feature in the DXOS ecosystem called composer. What is composer and what does it enable?
- [x] In a demo of composer you had this notion like app framework where everything inside the app was a plugin. Could you describe this environment a little more and what the plugin system is like?
- [x] One of the main goals of composer is to allow users to share data across apps. How does that look in practice?

**Comment by hipstersmoothie** [comment]
https://github.com/microsoft/inshellisense
https://sshx.io/

**Comment by just-be-dev** [comment]
- https://platform.openai.com/docs/guides/text-to-speech
- https://github.com/dflex-js/dflex

**Comment by jessmartin** [comment]
https://notes.jessmart.in <- my research notebook

real goal: discover and disseminate generalizable insights
